File: problems/2130_maximum_twin_sum_of_a_linked_list/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    int pairSum(ListNode* head) {
        vector<int> values;
        ListNode* curr = head;
        while (curr) { values.push_back(curr->val); curr = curr->next; }
        int n = values.size(), maxSum = 0;
        for (int i = 0; i < n / 2; i++) {
            maxSum = max(maxSum, values[i] + values[n - 1 - i]);
        }
        return maxSum;
    }
};

File: problems/2130_maximum_twin_sum_of_a_linked_list/Solution.java
import java.util.*;

class Solution {
    public int pairSum(ListNode head) {
        List<Integer> values = new ArrayList<>();
        ListNode curr = head;
        while (curr != null) { values.add(curr.val); curr = curr.next; }
        int n = values.size(), max = 0;
        for (int i = 0; i < n / 2; i++) {
            max = Math.max(max, values.get(i) + values.get(n - 1 - i));
        }
        return max;
    }
}
